scripts/docgen.mjs

```diff
@@ -0,0 +1,86 @@
+import fs from 'fs';
+import fetch from 'node-fetch';
+import jsdoc from 'jsdoc-api';
+
+// Fill this in to test a response locally, with fetching.
+const STUB = ``;
+const HEADER = `---
+# NOTE: This file is auto-generated from 'scripts/docgen.mjs'
+# Do not make edits to it directly, they will be overwritten.
+
+layout: ~/layouts/MainLayout.astro
+title: Configuration Reference
+---
+
+To configure Astro, add an \`astro.config.mjs\` file to the root of your project.
+
+\`\`\`js
+export default /** @type {import('astro').AstroUserConfig} */ ({
+  // all options are optional; these values are the defaults
+  projectRoot: './',
+  public: './public/',
+  dist: './dist/',
+  src: './src/',
+  pages: './src/pages/',
+  renderers: [
+    '@astrojs/renderer-svelte',
+    '@astrojs/renderer-vue',
+    '@astrojs/renderer-react',
+    '@astrojs/renderer-preact',
+  ],
+  vite: {},
+});
+\`\`\`
+`;
+
+const FOOTER = ``;
+
+/**
+ * The simple demo does not rely on the TypeScript compiler API; instead, it parses the
+ * source file directly.  It uses the default parser configuration.
+ */
+export async function run() {
+    const inputBuffer = STUB || await fetch('https://raw.githubusercontent.com/withastro/astro/wip-docs-autogenerate/packages/astro/src/%40types/astro.ts').then(r => r.text());
```

Comment:
Not blocking for this PR, but would it make sense to distribute `@types/astro.ts` (or a `.d.ts` equivalent) via npm?  Would be nice if this could eventually be read locally from the docs' `node_modules`

Reply:
Possibly! I personally don't mind fetching directly from the source like this, since this command is always run manually it's pretty low-touch. If we moved this into CI and it gets run more often/regularly, then I could definitely see value in reading from disk to cut down on the number of things that can go wrong in CI.